src/php/binaries.rs

```diff
@@ -78,13 +78,12 @@ pub(crate) fn all() -> HashMap<PhpVersion, PhpBinary> {
 
 fn get_all() -> HashMap<PhpVersion, PhpBinary> {
     let mut binaries: HashMap<PhpVersion, PhpBinary> = HashMap::new();
-    let os_infos = os_info::get();
-    let os_version = os_infos.version();
-    let bigsur = "11.0.0";
 
     binaries_from_rymfony_env(&mut binaries);
 
-    if cfg!(not(target_os = "macos")) || (cfg!(target_os = "macos") && VersionCompare::compare_to(&os_version.to_string(), &bigsur, &CompOp::Lt).unwrap()) {
+    // We don't want to use these PATH by default on macos grater than 'big sur'
+    // Apple add deprecation to the default binary which throws an error in the 'php:list' command
+    if cfg!(not(target_os = "macos")) || !is_macos_big_sur() {
```

Comment:
I'm sure that you could simply use `if !is_macos_big_sur` and move the `cfg!()` to  the function directly. You could create tow versions of this function: one for linux and windows that just return `false`, and one for MacOS that returns the logic used to detect the OS. It's a bit faster and more consistent OS-wise 😉